optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -247,7 +247,7 @@ public ScoreDefinition buildScoreDefinition() {
                     + ") or bendableSoftLevelsSize (" + bendableSoftLevelsSize + ").");
         }
         if (scoreDefinitionClass != null) {
-            if (scoreDefinitionType != null || bendableHardLevelsSize != null || bendableSoftLevelsSize != null) {
```

Comment:
I don't understand why this is a redundant condition (at least with the code on master, I am seeing this on line 262, not on line 250, please "git pull --rebase" master to confirm.

Here's a config I believe should trigger this fail fast error and wouldn't without the change:

    <scoreDirectorFactory>
        <scoreDefinitionClass>MyCustomDef</>
        <bendableHardLevelsSize>4</>
    </>

Ps: if it's redundant, then the IllegalStateException should probably be adjusted to not mention bendableHardLevelsSize.